### DISCUSSION TITLE
Open Viewer Feature Request

### DISCUSSION THREAD

**Discussion by chrisbrejon** [romainaugier/OpenViewer#3] [Ideas]
Hello,

apres discussion avec Manu Quinto (Mathematic), je me permets de relayer ces demandes ici.

Interessant de noter que Manu a teste pas mal de viewers sur le marche : Rv, [djview](https://darbyjohnston.github.io/DJV/), [keyframe](https://zurbrigg.com/keyframe-mp), [jefecheck](http://jefecheck.jefecorp.com/), [mrviewer](https://mrviewer.sourceforge.io/)... 
Et que finalement celui qui correspond le + a ses besoins est Rv.

Voici les feature request : 
1- Gestion du color management claire : 
![image](https://user-images.githubusercontent.com/70026961/124113162-68d3d100-da6b-11eb-8586-c2068788e3ca.png)
2- retouche de colo à la volée pour tirer un peu tes valeurs
![image](https://user-images.githubusercontent.com/70026961/124113204-7426fc80-da6b-11eb-8ef5-965286e918f1.png)
3- OCIO (v1, ok) qui me semble assez fiable
![image](https://user-images.githubusercontent.com/70026961/124113237-7e48fb00-da6b-11eb-9660-79ec2cf99388.png)
4- la possibilité de faire des playlists, et donc limite un petit montage à la volée
5- les matte pour faire des black bands facilement j'aime bien
![image](https://user-images.githubusercontent.com/70026961/124113304-9456bb80-da6b-11eb-9fce-346b76f260d4.png)
6- plein de petite manip sur la source, en particulier le framerate (et l'aspect ratio)
![image](https://user-images.githubusercontent.com/70026961/124113353-a173aa80-da6b-11eb-878e-157171ea1582.png)
7- une gestion de la mise en cache claire et configurable
8- un layer selector qui fait tout à fait le job pour les encapsulés
![image](https://user-images.githubusercontent.com/70026961/124113426-b8b29800-da6b-11eb-983e-25f1f43b44b9.png)
9- un color inspector qui mériterait un peu plus, mais qui a le mérite d'être précis
![image](https://user-images.githubusercontent.com/70026961/124113469-c405c380-da6b-11eb-8dc0-bca482e199b2.png)
10- se balader dans l'image, zoomer, scrubber sur la timeline, c'est fluide et réactif
11- possibilité d'utiliser des LUT évidemment
![image](https://user-images.githubusercontent.com/70026961/124113531-d5e76680-da6b-11eb-9b35-0562572c5d2d.png)
12- support de l'audio
13- support d'une bonne majorité de formats
14- un temps d'ouverture rapide (s'il y a un défaut que je pouvais désigner a Rv c'est peut-être son temps d'ouverture)
15- la possibilité de le rendre API-able (pour qu'on puisse le scripter)
16- feature pour afficher des 360 latlong (comme mrviewer)
17- le support du prores serait apprécié (probleme de license ?)

Merci !

**Comment by romainaugier** [comment]
C'est noté, merci pour la feature list précise !

**Comment by chrisbrejon** [comment]
Voici quelques features request supplémentaires (de Dorian). Il y a pleins de features avant de pouvoir prétendre à ne serait-ce que talonner Rv. La liste est longue : 
- Rapide à lancer 
- Différents formats : mp4, exr, png, jpeg
- Switch de layers (AOV d'un exr)
- OCIOv2
- pouvoir lire plusieurs vidéos en parallèle (à préciser avec Dorian)
- pixel picker 
- lecture du son
- remote controllable (lire un même fichier vidéo, mais par deux PC différents, et les deux ont le contrôle de l’application)
- Les même raccourcis claviers que RV par défaut. 

À mes yeux, c’est déjà un boulot monstrueux avec cette liste. Je n’ai donc pas de cahier des charges à proposer autre que ce que fait déjà RV. 

Merci !
